From: Luigi Toscano <luigi.toscano@tiscali.it>
Date: Sat, 18 Mar 2017 23:16:28 +0100
Subject: Error page, correctly load the image (with a real URL)

Summary:
The local address to the image is injected into an HTML page (the
error page). This address needs to be a proper URL with file:///,
not just the absolute path to the image file - otherwise
KIO::SimpleJobPrivate::simpleJobInit complains with "Invalid URL".

Test Plan:
Compiles, and when trying to load an invalid address in Konqueror
with the KHTML part, the warning picture on the top-left corner is
now visible.

Reviewers: mpyne

Reviewed By: mpyne

Subscribers: mpyne, #frameworks

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D5104
---
 src/khtml_part.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/khtml_part.cpp b/src/khtml_part.cpp
index d5f8d7f..3561ada 100644
--- a/src/khtml_part.cpp
+++ b/src/khtml_part.cpp
@@ -1916,7 +1916,7 @@ void KHTMLPart::htmlError(int errorCode, const QString &text, const QUrl &reqUrl
 
     html.replace(QLatin1String("TITLE"), i18n("Error: %1 - %2", errorName, url));
     html.replace(QLatin1String("DIRECTION"), QApplication::isRightToLeft() ? "rtl" : "ltr");
-    html.replace(QLatin1String("ICON_PATH"), KIconLoader::global()->iconPath("dialog-warning", -KIconLoader::SizeHuge));
+    html.replace(QLatin1String("ICON_PATH"), QUrl::fromLocalFile(KIconLoader::global()->iconPath("dialog-warning", -KIconLoader::SizeHuge)).url());
 
     QString doc = QLatin1String("<h1>");
     doc += i18n("The requested operation could not be completed");
